pkg/api/controller_test.go

```diff
@@ -1502,11 +1502,22 @@ func TestController_ListBranchesHandler(t *testing.T) {
 		if len(results) != 2 {
 			t.Fatalf("expected 2 branches to return, got %d", len(results))
 		}
-		retReference := results[0]
-		const expectedID = "main2"
-		if retReference.Id != expectedID {
-			t.Fatalf("expected '%s' as the first result for the second page, got '%s' instead", expectedID, retReference.Id)
+		require.Equal(t, results[0].Id, "main3")
+		require.Equal(t, results[1].Id, "main5")
+
+		// List all branches
+		resp, err = clt.ListBranchesWithResponse(ctx, repo, &apigen.ListBranchesParams{
+			After:      apiutil.Ptr[apigen.PaginationAfter]("main1"),
+			Amount:     apiutil.Ptr[apigen.PaginationAmount](2),
+			ShowHidden: swag.Bool(true),
```

Comment:
I think we want a different set of changes to the tests:
- ListBranches should be tested both with and without ShowHidden.  Right now it is not clear where we test ShowHidden=false.
- All other branch tests can just pick one value of Hidden and stick with it.  There is no special behaviour for diff on a hidden branch, or indeed for any other on-branch operation.

Reply:
This is what the test does.
line 1496 - test with hidden=false
line 1509 - test with hidden=true
We have also other unit tests which cover the hidden flag in more variations

Reply:
Sorry, missed that this is an added test.  Thanks fo explaining!